src/query.rs

```diff
@@ -40,6 +70,20 @@ where
         }
     }
 
+    /// Borrow a query to allow for iteration.
```

Comment:
I would rather say that this borrows the world, not the query. (I have a commit that I want to push after this PR that renames this lifetime to `'w` because it is how long the world stays borrowed.)